### ISSUE TITLE
Issue with Composer 2 PreFileDownloadEvent

### ISSUE THREAD

**Issue by emodric** [symfony/thanks#87]
After https://github.com/composer/composer/commit/5c13c974286969e4c5298e74c1a3ee66ef6e2bef, `symfony/thanks` breaks with the following error:

```
PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function Composer\Plugin\PreFileDownloadEvent::__construct(),
3 passed in /var/www/html/site/vendor/symfony/thanks/src/GitHubClient.php on line 191
and at least 4 expected in phar:///home/eddie/bin/composer2/src/Composer/Plugin/PreFileDownloadEvent.php:54
```

I'm not sure what the 4th parameter should be here? The commit above uses `package` and `metadata`. Maybe `metadata`?

Maybe @Seldaek could tell more.

**Comment by Seldaek** [comment]
Ah I guess we can make this optional.. but also I am not sure why symfony/thanks is triggering that event at all.

It probably should use its own "symfony/thanks" value there. Plugins probably shouldn't react to things they don't expect/know.

**Comment by nicolas-grekas** [comment]
It's triggering just in case a plugin wants to do something before/after/instead any downloads...
We could drop this, it wouldn't change anything from "thanks"' pov.

**Comment by emodric** [comment]
@nicolas-grekas Wanna a PR removing the event dispatch?

**Comment by nicolas-grekas** [comment]
yes, please!

**Comment by emodric** [comment]
@nicolas-grekas https://github.com/symfony/thanks/pull/88

**Comment by nicolas-grekas** [comment]
Fix released as v1.2.9

**Comment by emodric** [comment]
Thanks @nicolas-grekas @Seldaek !
